File: packages/python-sdk/src/treedb_sdk/adapters/search_index.py
from __future__ import annotations

from typing import Any, Mapping

from .common import json_request, segment
from treedb_sdk.transport import Transport


class SearchIndexAdapter:
    def __init__(self, transport: Transport) -> None:
        self.transport = transport

    def status(self, repo_id: str, query: Mapping[str, str | int | float | bool | None] | None = None) -> Any:
        return json_request(self.transport, "GET", f"/api/v1/repos/{segment(repo_id)}/search/index/status", query=query)

    def refresh(self, repo_id: str, body: Any | None = None) -> Any:
        return json_request(self.transport, "POST", f"/api/v1/repos/{segment(repo_id)}/search/index/refresh", body)

    def compact(self, repo_id: str, body: Any | None = None) -> Any:
        return json_request(self.transport, "POST", f"/api/v1/repos/{segment(repo_id)}/search/index/compact", body)

File: packages/elixir-sdk/lib/tree_db_sdk/adapters/search_index.ex
defmodule TreeDbSdk.SearchIndex do
  @moduledoc false
  alias TreeDbSdk.Adapters.Common

  def status(client, repo_id, query \\ %{}),
    do:
      Common.json_request(
        client,
        :get,
        "/api/v1/repos/" <> Common.segment(repo_id) <> "/search/index/status",
        nil,
        query
      )

  def refresh(client, repo_id, body \\ %{}),
    do:
      Common.json_request(
        client,
        :post,
        "/api/v1/repos/" <> Common.segment(repo_id) <> "/search/index/refresh",
        body
      )

  def compact(client, repo_id, body \\ %{}),
    do:
      Common.json_request(
        client,
        :post,
        "/api/v1/repos/" <> Common.segment(repo_id) <> "/search/index/compact",
        body
      )
end
